Source/Scene/MetadataProperty.js

```diff
@@ -0,0 +1,292 @@
+import Check from "../Core/Check.js";
+import clone from "../Core/clone.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import MetadataType from "./MetadataType.js";
+
+/**
+ * A metadata property.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String} options.id The ID of the property.
+ * @param {Object} options.property The property JSON object.
+ * @param {Object.<String, MetadataEnum>} [options.enums] A dictionary of enums.
+ *
+ * @alias MetadataProperty
+ * @constructor
+ *
+ * @private
+ */
+function MetadataProperty(options) {
```

Comment:
If we decide to have a `MetadataTableProperty`, this should be called `MetadataClassProperty` to make this clear.

Reply:
I renamed this file and made a few changes so that `MetadataTable` is a more manageable file:

* Added `MetadataTableProperty` for handling setting/getting individual properties. Did not add unit tests for this file because it is fully covered by the `MetadataTable` tests.
* Moved `MetadataTableProperty` `setProperty` validation code to `MetadataClassProperty`